BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonSource.java

```diff
@@ -0,0 +1,51 @@
+package com.hubspot.baragon.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BaragonSource {
+  private final String name;
+  private final Optional<Integer> port;
+
+  @JsonCreator
```

Comment:
Easier to ensure backwards compatibility via a static creator via:

```java
@JsonCreator
public static BaragonSource fromString(String value) {
    return new BaragonSource(value, Optional.absent());
}
```

This approach is nice because it'd remove the need for the ZK migration (i believe).